### PR TITLE
implements new login flow

### PR DESCRIPTION
**Pull request by mahabaleshwarhnr**
- implements cm_app_38(login with OTP)
- implements cm_app_53 (fetching login mode)

### CODE REVIEW

File: user/registration/build.gradle [hunk 1]
```diff
@@ -14,4 +16,7 @@ android {
 
 dependencies {
     api project(Modules.core)
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
+    implementation 'androidx.lifecycle:lifecycle-extensions:2.2.0'
```
Comment (AnusreeSajeevan) [inline user/registration/build.gradle]: Check if we can add these dependencies in deps.kt

File: user/registration/src/main/java/in/projecteka/jataayu/registration/ui/activity/LoginActivity.kt [hunk 2]
```diff
@@ -31,15 +32,17 @@ class LoginActivity : BaseActivity<ActivityLoginBinding>() {
...
 
+
     private fun initObservers() {
         viewModel.onClickRegisterEvent.observe(this, Observer {
             startRegistration(this)
         })
 
         viewModel.onPasswordVisibilityToggleEvent.observe(this, Observer {
-            binding.etPassword.setSelection(it)
+//            binding.etPassword.setSelection(it)
```
Comment (AnusreeSajeevan) [inline user/registration/src/main/java/in/projecteka/jataayu/registration/ui/activity/LoginActivity.kt]: Remove commented code

File: user/registration/src/main/java/in/projecteka/jataayu/registration/ui/fragment/ConsentManagerIDInputFragment.kt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        viewModel.onRegisterButtonClickEvent.observe(viewLifecycleOwner, Observer {
+
+        })
+
+        viewModel.onForgetCMIDButtonClickEvent.observe(viewLifecycleOwner, Observer {
+
+        })
+
+        viewModel.onNextButtonClickEvent.observe(viewLifecycleOwner, Observer {
+            // TODO - should remove after API integration
```
Comment (AnusreeSajeevan) [inline user/registration/src/main/java/in/projecteka/jataayu/registration/ui/fragment/ConsentManagerIDInputFragment.kt]: Remove this if api is integrated

File: user/registration/src/main/res/layout/consent_manager_id_input_fragment.xml [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+            android:padding="@dimen/content_margin_medium"
+            >
+
+            <TextView
+                android:id="@+id/tv_welcome"
+                style="@style/TextAppearance.Light"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="@dimen/content_margin_extra_extra_large"
+                android:text="welcome"
```
Comment (AnusreeSajeevan) [inline user/registration/src/main/res/layout/consent_manager_id_input_fragment.xml]: move hard coded strings and dimens to respective files

File: user/registration/src/main/res/layout/consent_manager_id_input_fragment.xml [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+                        android:layout_width="wrap_content"
+                        android:layout_height="match_parent"
+                        android:layout_gravity="end"
+                        android:drawablePadding="@dimen/content_margin_medium"
+                        android:text="@string/cm_config_provider"
+                        android:textColor="@color/black"
+                        android:textSize="@dimen/text_size_medium" />
+                    <ImageView
+                        android:layout_width="24dp"
+                        android:layout_height="24dp"
```
Comment (AnusreeSajeevan) [inline user/registration/src/main/res/layout/consent_manager_id_input_fragment.xml]: use values from dimens.xml

File: user/registration/build.gradle (hunk 1 above)
Comment (mahabaleshwarhnr) [reply user/registration/build.gradle]: comments are removed in latest commit

**Review by AnusreeSajeevan — APPROVED** [review]
Looks good

**Review by AnusreeSajeevan — APPROVED** [review]
Looks good

**Pull request merged**